crates/ruff_python_formatter/src/other/parameter_with_default.rs

```diff
@@ -18,7 +20,51 @@ impl FormatNodeRule<ParameterWithDefault> for FormatParameterWithDefault {
 
         if let Some(default) = default {
             let space = parameter.annotation.is_some().then_some(space());
-            write!(f, [space, token("="), space, group(&default.format())])?;
+            // ```python
+            // def f(
+            //     a = # parameter trailing comment; needs line break
+            //     1,
+            //     b =
+            //     # default leading comment; needs line break
+            //     2,
+            //     c = ( # the default leading can only be end-of-line with parentheses; no line break
+            //         3
+            //     ),
+            //     d = (
+            //         # own line leading comment with parentheses; no line break
+            //         4
+            //     )
+            // )
+            // ```
+            let needs_line_break_trailing = f.context().comments().has_trailing(parameter);
+            let default_first_comment = f.context().comments().leading(default.as_ref()).first();
+            let needs_line_break_leading = default_first_comment.is_some_and(|default_leading_comment| {
+                let mut tokenizer = SimpleTokenizer::new(
+                    f.context().source(),
+                    TextRange::new(parameter.end(), default_leading_comment.start()),
+                )
+                .skip_trivia()
+                .skip_while(|token| token.kind == SimpleTokenKind::RParen);
+                let equals = tokenizer.next();
+                debug_assert!(equals.is_some_and(|token| token.kind == SimpleTokenKind::Equals));
+                let lparens = tokenizer.next();
+                debug_assert!(lparens
+                    .as_ref()
+                    .map_or(true, |token| token.kind == SimpleTokenKind::LParen));
+                lparens.is_none()
+            });
+            let needs_line_break = needs_line_break_trailing || needs_line_break_leading;
+
+            write!(
+                f,
+                [
+                    space,
+                    token("="),
+                    (!needs_line_break).then_some(space),
+                    needs_line_break.then_some(hard_line_break()),
```

Comment:
This was meant as an if-then-else, not sure if it's better to split the `write!` into parts